File: techguidehub-backend/src/main/java/com/techguidehub/model/Post.java
package com.techguidehub.model;

import jakarta.persistence.*;


@Entity
@Table( name = "Posts")
public class Post {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long postId;
    private String title;
    private String body;
    private String type;

    public Long getPostId() {
        return postId;
    }
    public void setPostId(Long postId) {
        this.postId = postId;
    }
    public String getTitle() {
        return title;
    }
    public void setTitle(String title) {
        this.title = title;
    }
    public String getBody() {
        return body;
    }
    public void setBody(String body) {
        this.body = body;
    }
    public String getType() {
        return type;
    }
    public void setType(String type) {
        this.type = type;
    }


    @ManyToOne
    @JoinColumn( name = "userId")
    private User user;

    public User getUser() {
        return user;
    }
    public void setUser(User user) {
        this.user = user;
    }



    @ManyToOne
    @JoinColumn( name = "categoryId")
    private Category category;

    public Category getCategory() {
        return category;
    }
    public void setCategory(Category category) {
        this.category = category;
    }

    




}

File: frontend/src/app/models/post.ts
import { Category } from "./category";
import { User } from "./user";


export class Post {
  id?: number;
  title?: string;
  body?: string;
  dateCreated?: Date;
  likes?: number;
  user?: User;
  category?: Category;
  

  constructor(id?: number, title?: string, body?: string, dateCreated?: Date, likes?: number, user?: User) {
    this.id = id;
    this.title = title;
    this.body = body;
    this.dateCreated = dateCreated;
    this.likes = likes;
    this.user = user;
    
  }
}
